### PR TITLE
Initial implementations of Workspaces and Settings

### PR DESCRIPTION
**Pull request by kabentley**
See [this learning article](https://github.com/iTwin/itwinjs-core/blob/workspaces/docs/learning/backend/Workspace.md#workspacecontainers) for overview.  Basically `WorkspaceContainers`s and `Settings` are the way that **applications** can permit users to customization their behavior (much like "workspaces" and "config files" in PowerPlatform).

Note:

- both are backend-only concepts.
- `Settings` are merely key/value pairs with a priority scheme and a naming convention. They may be loaded from JSON files, Workspace resources, or as stringified JSON stored elsewhere (e.g. in a "setting service")
- `WorkspaceContainer`s hold application data

### CODE REVIEW

File: ui/components-react/src/components-react/toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -67,6 +67,7 @@ export function Toolbar(props: ToolbarProps) {
     });
   }, []);
 
+  // eslint-disable-next-line deprecation/deprecation
```
[the remark this answers is not in the document]
Comment (kabentley) [reply ui/components-react/src/components-react/toolbar/Toolbar.tsx]: @bsteinbk the doc problem is going to take some time to fix. In the meantime, can we lock the version of react?

[the remark this answers is not in the document]
Comment (bsteinbk) [reply ui/components-react/src/components-react/toolbar/Toolbar.tsx]: I already pushed a fix for the deprecation issue to this branch.  So, I don't think we need to lock the version of react but if you want to you can.  I have been looking at it and here are the only package changes I found between a build the creates docs and one that doesn't.
 loader-utils: 2.0.0            	2.0.1
 @itwin/itwinui-css': 0.37.1 	0.37.2
 @types/react': 17.0.31      	17.0.33
 azurite: 3.14.2                 	3.14.3
 redux: 4.1.1                     	4.1.2
 axe-core: 4.3.3 			4.3.5
 '@azure/core-client': 1.3.1    1.3.2
 core-js-compat: 3.18.3          3.19.0
 core-js-pure: 3.18.3               3.19.0

[the remark this answers is not in the document]
Comment (bsteinbk) [reply ui/components-react/src/components-react/toolbar/Toolbar.tsx]: I also see '@itwin/core-bentley': ^3.0.0-extension.1 in the bad yaml file and @itwin/core-bentley': 3.0.0-dev.96 in one that produces docs

**Comment by pmconne** [comment]
What is the relationship between these settings and the settings stored in the product settings service?

**Comment by kabentley** [comment]
> What is the relationship between these settings and the settings stored in the product settings service?

see [this article](https://github.com/iTwin/itwinjs-core/blob/workspaces/docs/learning/backend/Workspace.md) for explanation of "these settings". But, basically `Workspaces` and `Settings` are the way that **applications** can permit users to customization their behavior (much like "workspaces" and "config file" in PowerPlatform).

Note: 
 - both are backend-only concepts. 
 - `Settings` are merely key/value pairs with a priority scheme. They may be loaded from JSON files, Workspace resources, or as stringified JSON stored elsewhere (e.g. in a "setting service")
 - WorkspaceContainers hold application data

**Comment by calebmshafer** [comment]
> What is the relationship between these settings and the settings stored in the product settings service?

@pmconne the settings that are currently stored in product settings service (PDSS), and are the current IModelApp.settings, will be only available as User preferences going forward and not as any shared preferences.

@kabentley looks like I was too slow here but I did want to talk about what we're going to do about information the frontend needs as part of settings. Things like map layer sources and other information that's stored there today. The current backend-only approach won't cover that case as far as I could tell with Workspaces and Settings.

**Pull request merged**